src/integration-tests/src/showcase.rs

```diff
@@ -74,7 +87,191 @@ pub async fn run() -> Result<()> {
     assert_eq!(response.binding_uri, "/v1beta1/repeat:body");
     assert_eq!(response.request, Some(request));
 
-    let request = RepeatRequest::new()
+    Ok(())
+}
+
+async fn repeat_data_body(client: &showcase::client::Compliance) -> Result<()> {
+    let request = new_request();
+    let response = client
+        .repeat_data_body()
+        .with_request(request.clone())
+        .send()
+        .await?;
+    tracing::info!("response: {response:?}");
+    assert_eq!(response.binding_uri, "/v1beta1/repeat:body");
+    assert_eq!(response.request, Some(request));
+
+    Ok(())
+}
+
+async fn repeat_data_body_info(client: &showcase::client::Compliance) -> Result<()> {
+    let request = new_request();
+    let response = client
+        .repeat_data_body_info()
+        .with_request(request.clone())
+        .send()
+        .await?;
+    tracing::info!("response: {response:?}");
+    assert_eq!(response.binding_uri, "/v1beta1/repeat:bodyinfo");
+    assert_eq!(response.request, Some(request));
+    Ok(())
+}
+
+async fn repeat_data_query(client: &showcase::client::Compliance) -> Result<()> {
+    let request = new_request();
+    let response = client
+        .repeat_data_query()
+        .with_request(request.clone())
+        .send()
+        .await?;
+    tracing::info!("response: {response:?}");
+    assert_eq!(response.binding_uri, "/v1beta1/repeat:query");
+    let got = workaround_bug_2198(response);
+    assert_eq!(got, request);
+    Ok(())
+}
+
+async fn repeat_data_simple_path(client: &showcase::client::Compliance) -> Result<()> {
+    let request = new_request();
+    let response = client
+        .repeat_data_simple_path()
+        .with_request(request.clone())
+        .send()
+        .await?;
+    tracing::info!("response: {response:?}");
+    assert!(
+        response.binding_uri.starts_with("/v1beta1/repeat/"),
+        "{}",
+        response.binding_uri
+    );
+    assert!(
+        response.binding_uri.ends_with(":simplepath"),
+        "{}",
+        response.binding_uri
+    );
+    let got = workaround_bug_2198(response);
+    assert_eq!(got, request);
+    Ok(())
+}
+
+async fn repeat_data_path_resource(client: &showcase::client::Compliance) -> Result<()> {
+    let mut request = new_request();
+    request.info = request.info.map(|i| i.set_f_string("first/f-string-value"));
+    let response = client
+        .repeat_data_path_resource()
+        .with_request(request.clone())
+        .send()
+        .await?;
+    tracing::info!("response: {response:?}");
+    assert_eq!(
+        response.binding_uri,
+        "/v1beta1/repeat/{info.f_string=first/*}/{info.f_child.f_string=second/*}/bool/{info.f_bool}:pathresource"
```

Comment:
(for my understanding): we are sending a request to some path where all the substitutions were made, and then gapic-showcase is telling us what the generic path template looked like?

So if, let's say, the library picked the wrong path for this request, the assertion would fail.

Reply:
Yes. Presumably at some point we would test different bindings using calls with different variables set the same variables matching different templates.